src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/NodePropertyFilter.cs

```diff
@@ -182,7 +184,19 @@ public string ToPropertyFilterString(Type type, ValidForFlag validForFlag)
                     object propertyValue = value;
                     if (Type == typeof(string))
                     {
-                        propertyValue = $"'{propertyValue}'";
+                        var escapedString = CUtils.EscapeStringSQuote((string)propertyValue);
```

Comment:
Please make sure you call out when you do other fixes like this so they can be tracked and verified separately